game/gbtd_gbr0.ksy

```diff
@@ -0,0 +1,300 @@
+doc-ref: https://www.devrs.com/gb/hmgd/gbtd.html
+
+seq:
+  - id: magic
+    contents: GBO0
+  - id: objects
+    type: gbr0_object
+    repeat: eos
+
+types:
+  gbr0_object:
+    seq:
+      - id: object_type
+        type: u2
+        enum: type
+      - id: object_id
+        type: u2
+      - id: record_length
+        type: u4
+      - id: body
+        size: record_length
+        type:
+          switch-on: object_type
+          cases:
+            type::producer:      producer
+            type::tile_data:     tile_data
+            type::tile_settings: tile_settings
+            type::tile_export:   tile_export
+            type::tile_import:   tile_import
+            type::palettes:      palettes
+            type::tile_pal:      tile_pal
+    
+    enums:
+      type:
+        0x0001: producer
+        0x0002: tile_data
+        0x0003: tile_settings
+        0x0004: tile_export
+        0x0005: tile_import
+        0x000D: palettes
+        0x000E: tile_pal
+        0x00FF: deleted
+
+    types:
+      producer:
+        seq:
+          - id: name
+            type: strz
+            size: 30
+          - id: version
+            type: strz
+            size: 10
+          - id: info
+            type: strz
+            size: 80
+
+      tile_data:
+        seq:
+          - id: name
+            type: strz
+            size: 30
+          - id: width
+            type: u2
+          - id: height
+            type: u2
+          - id: count
+            type: u2
+          - id: color_set
+            size: 4
+          - id: tiles
+            size: width*height
+            repeat: expr
+            repeat-expr: count
+
+      tile_settings:
+        enums:
+          split_order:
+            0: left2right_top2bottom
+            1: top2bottom_left2right
+
+          color_set:
+            0: pocket
+            1: gameboy
+            2: gbc
+            3: sgb
+
+        types:
+          display_settings:
+            seq:
+              - id: grid_shown
+                type: b1
+              - id: nibble_markers_shown
+                type: b1
+
+        seq:
+          - id: tile_id
+            type: u2
+          - id: simple
+            type: u1
+          - id: flags
+            type: display_settings
+          - id: left_color
+            type: u1
+          - id: right_color
+            type: u1
+          - id: split_width
+            type: u2
+          - id: split_height
+            type: u2
+          - id: split_order
+            type: u1 # Note: Official specification incorrectly says "long"
+            enum: split_order
+          - id: color_set
+            type: u1
+            enum: color_set
+          - id: bookmarks
+            type: u2
+            repeat: expr
+            repeat-expr: 3
+          - id: auto_update
+            type: u1
+        
+      tile_export:
+        enums:
+          compression:
+            0x00: none
+            0x01: gb_compress
```

Comment:
We have expanded enums:

```yaml
0x01:
  - id: gb_compress
    doc-ref: .... # it'd be nice to docref to compression format and maybe even write a decompressor that can be used via process: . Some examples you can see in https://github.com/kaitai-io/kaitai_compress and a simpler one and in QSP PR in this repo
```

Reply:
The available docs is very sparse and there's a lot of link rot going on but I added a link to the Delphi source code of the algorithm. I def want to write a decompressor at some point but it's not top prio right now.